### PR TITLE
Add build tag for samples

### PR DESCRIPTION
**Pull request by hzyi-google**
Add `// +build sample` build tag to samples, so that running `go install ./...` from client library rood directory will ignore the samples. We will have sample-tester to test the samples so it is safe to do that.

### CODE REVIEW

**Comment by hzyi-google** [comment]
cc/ @googleapis/samplegen

File: internal/gensample/gensample.go [hunk 1]
```diff
@@ -233,6 +233,7 @@ func (g *generator) commit(gofmt bool, year int) ([]byte, error) {
 
 	var file bytes.Buffer
 	fmt.Fprintf(&file, license.Apache, year)
+	fmt.Fprintf(&file, "// +build sample\n\n")
```
Comment (noahdietz) [inline internal/gensample/gensample.go]: By convention, build constraints are typically placed at the top of the file on the first line with a new line following. I know the docs don't say this specifically, but it is the common convention and what you will find in google-cloud-go

Comment (hzyi-google) [reply internal/gensample/gensample.go]: Got it. Will update it shortly.

**Comment by hzyi-google** [comment]
I don't think so. I tried locally and it worked. I could have missed something for sure, but I think we can solve issues when we get one.

**Pull request merged**